### PR TITLE
Remove redundant error messages for report actions

### PR DESCRIPTION
**Pull request by therealsujitk**
### Details

This PR makes the following changes.

- Set the `error` fields in `failureData` to `{ [timestamp]: null }` to set an error with no message (removing redundant error messages).
- Modify `OfflineWithFeedback` to show only valid error messages.
    ```js
    const errorMessages = _.omit(props.errors, e => e === null);
    const hasErrorMessages = !_.isEmpty(errorMessages);
    ```

### Fixed Issues

$ https://github.com/Expensify/App/issues/18839
PROPOSAL: https://github.com/Expensify/App/issues/18839#issuecomment-1546837018

### Tests

1. Log in with the same account in the main testing device and a secondary device
2. Disable the internet connection in the main testing device (Force Offline)
3. In the secondary device Split a bill with 2 users, this should create a new group and 2 new 1:1 conversations
4. In the main testing device (should be offline) create the SAME split bill with the 2 users
5. Enable the internet connection in the main testing device. The split bill request will be sent and you will get an error in the main device only
6. Duplicated chats are display in the LHN
7. Open the group chat report, 1:1 chat reports & 1:1 IOU reports and verify that there are no redundant error messages.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Log in with the same account in the main testing device and a secondary device
2. Disable the internet connection in the main testing device (Force Offline)
3. In the secondary device Split a bill with 2 users, this should create a new group and 2 new 1:1 conversations
4. In the main testing device (should be offline) create the SAME split bill with the 2 users
5. Enable the internet connection in the main testing device. The split bill request will be sent and you will get an error in the main device only
6. Duplicated chats are display in the LHN
7. Open the group chat report, 1:1 chat reports & 1:1 IOU reports and verify that there are no redundant error messages.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-06-15 at 3 26 13 PM

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by therealsujitk** [comment]
Few things I'd like to point out,

- While testing this PR I ran into a bug that I've reported on [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1686655622360139). This might be a regression as I don't remember this happening before (reproducible in production).
    image

- Normally in chat reports the error for the report in displayed in the create report action, however this action is not displayed for IOU reports, therefore currently no error is shown if the IOU report failed to create.
    image

File: src/components/OfflineWithFeedback.js [hunk 1]
```diff
@@ -89,6 +89,8 @@ function applyStrikeThrough(children) {
 
 const OfflineWithFeedback = (props) => {
     const hasErrors = !_.isEmpty(props.errors);
+    const errorMessages = _.omit(props.errors, (e) => e === null);
+    const hasErrorMessages = !_.isEmpty(errorMessages);
```
Comment (s77rt) [inline src/components/OfflineWithFeedback.js:93]:
```suggestion
    const hasErrorMessages = hasErrors && _.some(props.errors);
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
@therealsujitk Please double check if those errors are a regression or not before reporting in Slack. If they are, we should fix them here.

**Comment by therealsujitk** [comment]
@s77rt the bug I mentioned above isn't caused by this PR. Both points I mentioned can be considered out of scope.

File: src/components/OfflineWithFeedback.js (hunk 1 above)
Comment (s77rt) [reply src/components/OfflineWithFeedback.js:93]: Why would be pass `null` errors in that case?

Comment (therealsujitk) [reply src/components/OfflineWithFeedback.js:93]: I wanted to avoid the possibility of blank lines in case at some point in future the object has both valid and `null` errors.

Comment (s77rt) [reply src/components/OfflineWithFeedback.js:93]: Ah right nice catch. Can you just add a comment explaining that we may add errors with null to apply opacity without showing errors

Comment (therealsujitk) [reply src/components/OfflineWithFeedback.js:93]: Sure, but where should I add it?

Comment (s77rt) [reply src/components/OfflineWithFeedback.js:93]: just above `errorMessages`

File: src/components/OfflineWithFeedback.js [hunk 2]
```diff
@@ -89,6 +89,9 @@ function applyStrikeThrough(children) {
 
 const OfflineWithFeedback = (props) => {
     const hasErrors = !_.isEmpty(props.errors);
+    // Some errors might not have error messages when we want to apply opacity without showing any errors.
```
Comment (s77rt) [inline src/components/OfflineWithFeedback.js:92]:
```suggestion

    // Some errors have a null message. This is used to apply opacity only and to avoid showing redundant messages.
```
maybe this is better?

File: src/libs/actions/IOU.js [hunk 3]
```diff
@@ -237,6 +241,9 @@ function buildOnyxDataForMoneyRequest(
                           [iouCreatedAction.reportActionID]: {
                               errors: ErrorUtils.getMicroSecondOnyxError('iou.error.genericCreateFailureMessage'),
                           },
+                          [iouAction.reportActionID]: {
+                              errors: ErrorUtils.getMicroSecondOnyxError(null),
+                          },
```
Comment (s77rt) [inline src/libs/actions/IOU.js:246]: Can you please state where the `iouAction.reportActionID` errors field is being set?

File: src/libs/actions/IOU.js [hunk 4]
```diff
@@ -957,7 +966,7 @@ function getSendMoneyParams(report, amount, currency, comment, paymentMethodType
         optimisticReportActionsData.value[optimisticCreatedAction.reportActionID] = optimisticCreatedAction;
 
         // If we're going to fail to create the report itself, let's not have redundant error messages for the IOU
-        failureData[0].value[optimisticIOUReportAction.reportActionID] = {pendingAction: null};
+        failureData[0].value[optimisticIOUReportAction.reportActionID] = {errors: ErrorUtils.getMicroSecondOnyxError(null)};
```
Comment (s77rt) [inline src/libs/actions/IOU.js:969]: Same not as above. Do not overwrite but set the errors field correctly at the first initialisation. 
Also we are no longer clearing the pendingAction is this change required?

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/actions/IOU.js [hunk 5]
```diff
@@ -221,11 +221,15 @@ function buildOnyxDataForMoneyRequest(
...
                       }
-                    : {}),
-                [reportPreviewAction.reportActionID]: {
-                    created: reportPreviewAction.created,
-                },
+                    : {
+                          [reportPreviewAction.reportActionID]: {
+                              created: reportPreviewAction.created,
+                          },
+                      }),
```
[the remark this answers is not in the document]
Comment (therealsujitk) [reply src/libs/actions/IOU.js:232]: This is done so we can set the null error to the report action in the if condition. We need this for the opacity.

File: src/libs/actions/IOU.js [hunk 6]
```diff
@@ -221,11 +221,15 @@ function buildOnyxDataForMoneyRequest(
                           [chatCreatedAction.reportActionID]: {
                               errors: ErrorUtils.getMicroSecondOnyxError('iou.error.genericCreateFailureMessage'),
                           },
+                          [reportPreviewAction.reportActionID]: {
+                              errors: ErrorUtils.getMicroSecondOnyxError(null),
+                          },
```
[the remark this answers is not in the document]
Comment (therealsujitk) [reply src/libs/actions/IOU.js:226]: There was no error before, we need this for the opacity effect.

I've only added this error if it's a new chat report and not for existing chat reports. The reason for this is because this particular report action is re-used whenever an IOU is made.

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (therealsujitk) [reply src/libs/actions/IOU.js:246]: I don't understand, but the reason we're setting this is so we have the opacity effect for the report action in the IOU report.

File: src/libs/actions/IOU.js (hunk 4 above)
Comment (therealsujitk) [reply src/libs/actions/IOU.js:969]: Clearing the `pendingAction` is not required if there's an error, this is done only when the API write has succeeded. This was done before just to remove redundant errors.

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (s77rt) [reply src/libs/actions/IOU.js:246]: Okay, so I understand that we had no errors at all? So this is not related to remove redundant error messages but to the missing opacity itself?

Comment (therealsujitk) [reply src/libs/actions/IOU.js:246]: Correct. Same for report preview action.

Comment (therealsujitk) [reply src/libs/actions/IOU.js:246]: This one has an error if the report is not new.

File: src/libs/actions/IOU.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (therealsujitk) [reply src/libs/actions/IOU.js:232]: I don't think we need to reset `created` back to the old value in the if condition since the report action preview will definitely be new if the chat is new.

[the remark this answers is not in the document]
Comment (s77rt) [reply src/libs/actions/IOU.js:232]: This preview report action is not necessary linked to the chat. We have another parameter `isNewReportPreviewAction`. I think we should reset this report action as follow:
`isNewReportPreviewAction ? null : {created: reportPreviewAction.created}`

[the remark this answers is not in the document]
Comment (therealsujitk) [reply src/libs/actions/IOU.js:232]: That parameter is for if the chat is not new and the report preview is newly created, for example if a split is created with existing members. If the chat is new then then the report preview is definitely new in this case.

We can't reset the report preview action to null as it's still required to render.

[the remark this answers is not in the document]
Comment (s77rt) [reply src/libs/actions/IOU.js:232]: or actually set an error instead of `null`

[the remark this answers is not in the document]
Comment (therealsujitk) [reply src/libs/actions/IOU.js:232]: So I'll reset created inside the if condition?

```js
{
    onyxMethod: Onyx.METHOD.MERGE,
    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
    value: {
        ...(isNewChatReport
            ? {
                    [chatCreatedAction.reportActionID]: {
                        errors: ErrorUtils.getMicroSecondOnyxError('iou.error.genericCreateFailureMessage'),
                    },
                    [reportPreviewAction.reportActionID]: {
                        created: reportPreviewAction.created,  // Added this
                        errors: ErrorUtils.getMicroSecondOnyxError(null),
                    },
                }
            : {
                    [reportPreviewAction.reportActionID]: {
                        created: reportPreviewAction.created,
                    },
              }),
    },
}
```

This would keep `created` the same as before.

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (s77rt) [reply src/libs/actions/IOU.js:246]: This does not look correct, can you add an error for this?

File: src/libs/actions/IOU.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (s77rt) [reply src/libs/actions/IOU.js:232]: Oh I see, nvm, do nothing. There is no point in resting the `created` to the current `created` field (same value)

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (therealsujitk) [reply src/libs/actions/IOU.js:246]: You mean because there's no error message anywhere? It wasn't there before as well (It's one of the bullet points - https://github.com/Expensify/App/pull/20666#issuecomment-1589127887). 

Normally the error message would come with the create action, but that is hidden for iou reports, I thought maybe this is a feature request and the design team should tell us where to put the error message. If you still want me to add an error message in this action instead of the null error let me know.

Comment (s77rt) [reply src/libs/actions/IOU.js:246]: Hmm it looks we already have an error set here https://github.com/Expensify/App/blob/78145cfdc3f8449f4c42db064890b7e0fb90553b/src/libs/actions/IOU.js#L202

Comment (therealsujitk) [reply src/libs/actions/IOU.js:246]: Correct, that is for the report, we have one for the create action too here, but this action is hidden in the frontend.

https://github.com/Expensify/App/blob/78145cfdc3f8449f4c42db064890b7e0fb90553b/src/libs/actions/IOU.js#L238

Comment (s77rt) [reply src/libs/actions/IOU.js:246]: I see, let's not worry about this then.

File: src/libs/actions/IOU.js [hunk 7]
```diff
@@ -943,21 +962,39 @@ function getSendMoneyParams(report, amount, currency, comment, paymentMethodType
...
-            key: optimisticChatReportData.key,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticIOUReport.reportID}`,
             value: {
-                errorFields: {
-                    createChat: ErrorUtils.getMicroSecondOnyxError('report.genericCreateReportFailureMessage'),
+                [optimisticIOUReportAction.reportActionID]: {
+                    errors: ErrorUtils.getMicroSecondOnyxError('iou.error.other'),
                 },
             },
         });
```
Comment (s77rt) [inline src/libs/actions/IOU.js:997]: I don't see why we are conditionally setting the error of the IOU report action based on the CHAT report.
Chat reports and IOU reports are not rendered together, so why we are making this change?

Comment (therealsujitk) [reply src/libs/actions/IOU.js:997]: I'm actually not sure about this one, I'll have to look at the code a bit more to see what's going on but this was the original code in `main`, I didn't add anything that was not necessary.

https://github.com/Expensify/App/blob/14e3e1b848ac7df4505d87685552cff8f7c7833f/src/libs/actions/IOU.js#L959-L960

This code was written by @yuwenmemon so maybe he can shed some light on this?

Comment (yuwenmemon) [reply src/libs/actions/IOU.js:997]: I added this so we could clear out all failure data for IOU, because the error for creating the report itself should supercede it here. You can check out the PR here: https://github.com/Expensify/App/pull/18328

Whether it was the correct way to go... 🤷‍♂️

Comment (therealsujitk) [reply src/libs/actions/IOU.js:997]: @s77rt it looks like this is for the iou report action in chats, that should answer your question.

Comment (therealsujitk) [reply src/libs/actions/IOU.js:997]: I haven't tested this case as I didn't have the steps before (in the PR linked), if possible can you test for it while creating the reviewer checklist since it's already day 2? I'll be able to test it only tomorrow morning (IST).

Comment (s77rt) [reply src/libs/actions/IOU.js:997]: I guess we can move on then, just please remove the redundant `optimisticIOUReportAction.reportActionID` `failureData` set above the if condition (it will be overwritten).

https://github.com/Expensify/App/blob/999cd0bb7052929a70db2cdb6d3b0386c39180e4/src/libs/actions/IOU.js#L932-L940

Comment (therealsujitk) [reply src/libs/actions/IOU.js:997]: Ah I forgot this. Done!

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



web


</details>

<details>
<summary>Mobile Web - Chrome</summary>


mweb-chrome


</details>

<details>
<summary>Mobile Web - Safari</summary>



mweb-safari


</details>

<details>
<summary>Desktop</summary>



desktop


</details>

<details>
<summary>iOS</summary>



ios


</details>

<details>
<summary>Android</summary>



android


</details>

**Comment by s77rt** [comment]
@therealsujitk Please add missing screenshots

**Comment by therealsujitk** [comment]
@s77rt I've added the screenshot for Desktop but I can't test on iPhone right now since I don't own a Mac and my hackintosh is not powerful enough to handle a simulator. I have already ordered a MacBook which should be arriving by the end of this month after which I will be able to upload tests with iPhone for future PRs. Sorry for the inconvenience.

**Review by s77rt — APPROVED** [review]
LGTM! :rocket: 

cc @yuwenmemon

**Review by yuwenmemon — APPROVED** [review]
